### ISSUE TITLE
Support for React 15.4.x?

### ISSUE THREAD

**Issue by struesdell** [remarkablemark/html-react-parser#29]
It looks like this library has been broken by the move to React 15.4.x.

````
ERROR in ./~/html-react-parser/lib/attributes-to-props.js
Module not found: Error: Cannot resolve module 'react/lib/HTMLDOMPropertyConfig' in MYPATH/node_modules/html-react-parser/lib
 @ ./~/html-react-parser/lib/attributes-to-props.js 6:28-70

ERROR in ./~/html-react-parser/lib/property-config.js
Module not found: Error: Cannot resolve module 'react/lib/HTMLDOMPropertyConfig' in MYPATH/node_modules/html-react-parser/lib
 @ ./~/html-react-parser/lib/property-config.js 7:28-70

ERROR in ./~/html-react-parser/lib/property-config.js
Module not found: Error: Cannot resolve module 'react/lib/SVGDOMPropertyConfig' in MYPATH/node_modules/html-react-parser/lib
 @ ./~/html-react-parser/lib/property-config.js 8:27-68
````

**Comment by struesdell** [comment]
Thanks for the quick response! NPM wouldn't let me roll back to 15.3 so this was a lifesaver!

**Comment by remarkablemark** [comment]
No problem! I'm still seeing the error in my webpack build for the client bundle, so I'm looking into why it's still not working there. The node package should be fine. Keep me updated if it still gives you any errors.

**Comment by struesdell** [comment]
Okay yeah, it looks like the first error re: `attributes-to-props.js` is gone, but I'm seeing the same errors with `DOMPropertyConfig.js`.

**Comment by remarkablemark** [comment]
This is when you webpack, right? I actually realized the error does not break anything because webpack will try to bundle all requires whether it exists or not.

In this case, since I use a [try-catch](https://github.com/remarkablemark/html-react-parser/pull/30/files#diff-a814947f0ab8e9e2686ad8b98bbef956R11) to require the configs from both `react-dom/lib/` and `react/lib/`, there will always be a missing set of modules. (In other words, it will find `react-dom/lib/...` but not find `react/lib/...` or find `react/lib/...` but not `react-dom/lib/...`.)

To remove the error, I can do one of the following:
- Save the actual react-dom configs in the repository (remove the hard dependency)
- Create an additional build step before webpack runs that copies the config to a temp location
- Bump the version of the package so that it expects peerDependencies `react` and `react-dom` to be **>=15.4** will only require the config from `react-dom`

**Comment by struesdell** [comment]
Yeah I'm seeing my app and parser use-cases are still fully functional after the errors.
I'm not too concerned about the immediate warnings, and do really appreciate the effort.

> Bump the version of the package so that it expects peerDependencies react and react-dom to be >=15.4 will only require the config from react-dom

This third options feels like the right solution for the long-term, but since 15.4 just hit yesterday I understand preferring to stay stable and avoid forcing upgrades for other consumers. 👍

**Comment by remarkablemark** [comment]
Thanks @struesdell for the feedback. I'll continue to think about the different solutions but I believe your recommendation makes the most sense. I'll most likely make the changes later today.

**Comment by remarkablemark** [comment]
@struesdell, as we've discussed, I've published `0.3.0`, which should be good for `react` and `react-dom` >=15.4 (#32)

`0.2.0` will support `react` <= 15.3 (#31)
